src/module/module.jl

```diff
@@ -187,13 +187,13 @@ end
 ###############################################################################
 
 @doc Markdown.doc"""
-    sres{T <: Nemo.RingElem}(I::smodule{T}, max_length::Int)
+    sres{T <: Singular.FieldElem}(I::smodule{T}, max_length::Int)
 
 Compute a free resolution of the given module $I$ of length up to the given
 maximum length. If `max_length` is set to zero, a full length free
 resolution is computed. Each element of the resolution is itself a module.
 """
-function sres(I::smodule{T}, max_length::Int) where T <: Nemo.RingElem
+function sres(I::smodule{T}, max_length::Int) where T <: Singular.FieldElem
```

Comment:
So now the only error is:
`MethodError: no method matching sres(::smodule{spoly{n_Q}}, ::Int64)`

This should be
`function sres(I::smodule{spoly{T}}, max_length::Int) where T <: Nemo.FieldElem`
with a
`sresolution{spoly{T}}(R, r, Bool(minimal))`
at the end.

Outside of a very limited number of cases, which I will make sure are documented soon, there isn't a reason to use `Singular.FieldElem`. It is a limitation of julia type system.